Automattic/sensei
# Add quiz progress bar and exit course blocks

Fixes #4644

### Changes proposed in this Pull Request

- Add a Quiz Progress Bar block
- Add a Progress_Bar block component shared between quiz and course progress blocks
- Use the new block in `Single_Quiz::the_quiz_progress_bar`, replacing template and frontend JS
- Add Exit Course block

### Note

In Learning mode, the progress bar is always shown, even if pagination is off for the quiz. Customizations in quiz settings (color, height etc) are also ignored.

### Testing instructions

* Open a quiz in learning mode and check out the quiz progress bar.
* Open a quiz in non-learning mode. Turn on pagination, enable and customize progress bar. Make sure it follows settings. (And is not shown when pagination is disabled)
* Check that the course progress block in the course page also still works
--
* Check Exit Course button in learning mode sidebar going to the course



### Screenshot / Video

<img width="1370" alt="image" src="https://user-images.githubusercontent.com/176949/150600345-4ebd19f8-a8d0-4a95-a9be-cd0c6209d159.png">
